snuba/datasets/dataset.py

```diff
@@ -45,7 +46,7 @@ class Dataset(object):
       query before deciding which Storage to use. These processors are defined
       by the dataset
     - the Storage to run the query onto is selected and the query is transformed
-      into a Storage Query. This is done by a StorageQueryPlanBuilder. This object
+      into a Storage Query. This is done by a ClickhouseQueryPlanBuilder. This object
```

Comment:
I made the dataset depend directly on ClickhouseQueryPlanBuilder instead of having an intermediate generic QueryPlanBuilder. If I made the dataset depend on an abstract query plan builder instead, the query processing logic (which is clickhosue specific) would have had to ensure the instance returned by the dataset is the clickhouse one with various isinstance. 
Considering we are not likely to have any non clickhouse storage any time soon (for quite some time), it seems that this dependency is less leaky after we decided physical layer = clickhouse for the foreseeable future.

Adding the abstract QueryPlanBuilder is an easy change. Let me know what you think.